network/dag/bbolt_payloadstore.go

```diff
@@ -19,43 +19,45 @@
 package dag
 
 import (
+	"github.com/nuts-foundation/go-stoabs"
 	"github.com/nuts-foundation/nuts-node/crypto/hash"
-	"go.etcd.io/bbolt"
+	"github.com/nuts-foundation/nuts-node/network/log"
 )
 
-// payloadsBucketName is the name of the Bolt bucket that holds the payloads of the transactions.
-const payloadsBucketName = "payloads"
+// payloadsShelf is the name of the shelf that holds the payloads of the transactions.
+const payloadsShelf = "payloads"
 
-// NewBBoltPayloadStore creates a etcd/bbolt backed payload store using the given database.
-func NewBBoltPayloadStore(db *bbolt.DB) PayloadStore {
-	return &bboltPayloadStore{db: db}
+// NewPayloadStore creates a etcd/bbolt backed payload store using the given database.
+func NewPayloadStore() PayloadStore {
+	return &payloadStore{}
 }
 
-type bboltPayloadStore struct {
-	db *bbolt.DB
-}
+type payloadStore struct{}
 
-func (store bboltPayloadStore) isPayloadPresent(tx *bbolt.Tx, payloadHash hash.SHA256Hash) bool {
-	bucket := tx.Bucket([]byte(payloadsBucketName))
-	if bucket == nil {
+func (store payloadStore) isPayloadPresent(tx stoabs.ReadTx, payloadHash hash.SHA256Hash) bool {
+	reader, err := tx.GetShelfReader(payloadsShelf)
+	if err != nil {
+		log.Logger().Errorf("failed to verify payload existence (hash=%s): %s", payloadHash, err)
 		return false
 	}
-	data := bucket.Get(payloadHash.Slice())
+	data, err := reader.Get(stoabs.NewHashKey(payloadHash))
 	return len(data) > 0
 }
 
-func (store bboltPayloadStore) readPayload(tx *bbolt.Tx, payloadHash hash.SHA256Hash) []byte {
-	bucket := tx.Bucket([]byte(payloadsBucketName))
-	if bucket == nil {
+func (store payloadStore) readPayload(tx stoabs.ReadTx, payloadHash hash.SHA256Hash) []byte {
+	reader, err := tx.GetShelfReader(payloadsShelf)
+	if err != nil {
+		log.Logger().Errorf("failed to read payload (hash=%s): %s", payloadHash, err)
 		return nil
 	}
-	return bucket.Get(payloadHash.Slice())
+	data, err := reader.Get(stoabs.NewHashKey(payloadHash))
+	return data
```

Comment:
error is ignored

Reply:
error returned (as described by the interface)